libraries/commerce/product/reducers/productsById.js

```diff
@@ -49,6 +50,20 @@ export default function productsById(state = {}, action) {
           isFetching: false,
         },
       };
+    case UPDATE_METADATA:
+      return {
```

Comment:
Perhaps it would be more prudent to construct this new data structure step by step, with hints, before returning it.

Reply:
done